pl_examples/domain_templates/imagenet.py

```diff
@@ -177,17 +176,28 @@ def val_dataloader(self):
             ])),
             batch_size=self.batch_size,
             shuffle=False,
-            num_workers=0,
+            num_workers=self.workers,
         )
         return val_loader
 
+    def test_dataloader(self, *args, **kwargs):
+        return self.val_dataloader(*args, **kwargs)
+
+    def test_step(self, *args, **kwargs):
+        return self.validation_step(*args, **kwargs)
+
+    def test_epoch_end(self, *args, **kwargs):
+        return self.validation_epoch_end(*args, **kwargs)
+
```

Comment:
It is not correct to directly redirect to validation methods here, because the metric names will refer to "val_...". This will affect the logging and progress bar display.
You can do it like this but you need to  fetch the output and replace the key names to "test_...".

Reply:
anyway, in v0.9 this will change with the new structured outputs:)

Reply:
@awaelchli mind edit it?

Reply:
The main reason I do this is to use the Trainer.test(). For imagenet, the evaluation is supposed to run on validation set, so the progress bar is fine.

Reply:
I just want to make sure this example is solid and does not create a misunderstanding of what test and eval is.
You can always pass in the val dataloader to Trainer.test(), but what I mean is that the logged plots will look weird if you run validation during training and then at the end also run test, which will append the logs to the validation results if the have the same names.

Reply:
I see. would trainer.test(model, test_dataloaders=model.val_dataloader()) be better?

Reply:
Something like this?

```
outputs = self.validation_epoch_end(*args, **kwargs)
outputs = {k.replace('val', 'test'):v for k,v in outputs.items()}
return outputs
```

Reply:
I pushed something.